lib/ansible/modules/net_tools/speedtest.py

```diff
@@ -0,0 +1,216 @@
+          - URL of a Speedtest Mini server
+        required: false
+    pre_allocate:
+        default: true
+        description:
+          - Pre-allocate upload data. This option improves performance of
+            upload testing, but comes at the cost of higher memory use
+        type: bool
+    validate_certs:
+        default: True
+        description:
+          - Use HTTPS instead of HTTP when communicating with
+            speedtest.net operated servers
+        type: bool
+    servers:
+        default: []
+        description:
+          - List of servers to limit for testing
+        required: false
+    share:
+        default: false
+        description:
+          - Generate and provide a URL to the speedtest.net share
+            results image
+        type: bool
+    source:
+        description:
+          - Source IP address to bind to
+        required: false
+    timeout:
+        default: 10
+        description:
+          - HTTP timeout in seconds
+    upload:
+        default: true
+        description:
+          - Perform upload test
+        type: bool
+requirements:
+  - speedtest-cli>=2.0.0
+version_added: 2.7.0
+'''
+
+EXAMPLES = '''
+- speedtest:
+    share: yes
+    servers:
+      - 1234
+
+- speedtest:
+    share: yes
+    source: "{{ ansible_eth1.ipv4.address }}"
+'''
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+RETURN = '''
+bytes_received:
+    description: bytes recevied during download testing
+    returned: always
+    type: int
+    sample: 315786858
+bytes_sent:
+    description: bytes sent during upload testing
+    returned: always
+    type: int
+    sample: 105914368
+client:
+    description: information about local client
+    returned: always
+    type: dict
+download:
+    description: download speed in bytes per second
+    returned: always
+    type: float
+    sample: 252432440.16133374
+ping:
+    description: latency to chosen server
+    returned: always
+    type: float
+    sample: 5.491
+server:
+    description: information about the remote test server
+    returned: always
+    type: dict
+share:
+    description: URL to speedtest.net share image, will be null if share=False
+    returned: always
+    type: str
+    sample: http://www.speedtest.net/result/1234.png
+timestamp:
+    description: ISO8601 timestamp recorded at the start of the test
+    returned: always
+    type: str
+    sample: "2018-04-02T16:00:20.036880Z"
+upload:
+    description: upload speed in bytes per second
+    returned: always
+    type: float
+    sample: 81690942.203669
+'''
+
+import traceback
+
+try:
+    import speedtest
+    HAS_SPEEDTEST = True
+except ImportError:
+    HAS_SPEEDTEST = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_speedtest(module):
+    s = speedtest.Speedtest(
+        source_address=module.params['source'],
+        secure=module.params['secure'],
```

Comment:
`secure` isn't defined as a module param

Reply:
Boo, I renamed right before I pushed to be `validate_certs` and apparently sucked at it.

Reply:
fixed